### PR TITLE
LOG-1375: implementation attempt

### PR DESCRIPTION
**Pull request by humzashah**
required for [LOG-1375](https://deliveroo.atlassian.net/browse/LOG-1375)

Adds all the code required for implementation. At the moment, this gem can be installed through GitHub. It will be moved to RubyGems once we decide to replace our current implementation with this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+]
+
+HungarianAlgorithmC.find_pairings(costs)
+
+#=> [[0, 0], [1, 2], [2, 1]]
+```
+
+## Acknowledgements
+
+The C code uses the implementation by [Cyrill Stachniss](ext/hungarian_algorithm_c/libhungarian).
```
Comment (mezis) [inline README.md:49]: If you don't want Legal to throw a fit — you probably want to mention that the code in question has been placed in the public domain by its author explicitly.
And link to the dude's [page](http://www2.informatik.uni-freiburg.de/~stachnis/misc.html)?

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+machine:
+  ruby:
+    version: 2.2.7
```
Comment (mezis) [inline circle.yml]: I'd suggest testing with multiple Ruby versions straight away.
Given CircleCI makes that [quite painful](https://circleci.com/docs/1.0/parallel-manual-setup/#build-matrix-support), maybe just use Travis?

File: hungarian_algorithm_c.gemspec [hunk 3]
```diff
@@ -0,0 +1,30 @@
+require_relative './lib/hungarian_algorithm_c/version'
+require 'date'
+
+Gem::Specification.new do |s|
+  s.name = 'hungarian_algorithm_c'
+  s.version = HungarianAlgorithmC::VERSION
+  s.date = Date.today.to_s
```
Comment (mezis) [inline hungarian_algorithm_c.gemspec]: Well... no, given the spec is "run" every time it's checked, the gem release date isn't "today" :)

Your gemspec looks really odd, too. Did you scaffold it with `bundle gem`? Or try to prep it manually?

File: hungarian_algorithm_c.gemspec [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  s.summary = 'Evaluates the Hungarian algorithm in C'
+  s.description = 'A Ruby gem that evaluates the Hungarian algorithm in C'
+  s.homepage = 'https://github.com/deliveroo/hungarian_algorithm_c'
+  s.license = 'MIT'
+
+  s.authors = ['Logistics Algorithms']
+  s.email = 'logistics+algorithms-engineers@deliveroo.co.uk'
```
Comment (mezis) [inline hungarian_algorithm_c.gemspec]: Conventionally authors are actual people. Why not put your name here?

**Review by mezis — APPROVED** [review]
Looks good, just a few minor gripes. Excited to see how this performs for you :+1:

File: README.md (hunk 1 above)
Comment (humzashah) [reply README.md:49]: I'll mention the public domain, but the page link is there. If you click on through the link here, I've added the person's details in a whole README.md dedicated to their part of the code. :)

File: circle.yml (hunk 2 above)
Comment (humzashah) [reply circle.yml]: Yes, I suppose could use Travis - not like we'll be changing this gem often.

File: README.md (hunk 1 above)
Comment (mezis) [reply README.md:49]: Ah, duh. Thanks.

**Pull request merged**